File: server-SpringMvc/enuocms-core/src/main/java/com/enuocms/core/model/PageResult.java
package com.enuocms.core.model;


import lombok.Data;

import java.util.ArrayList;
import java.util.List;


/**
 * Created by zhanxiaoping on 2017/8/29.
 * zhanxp@me.com
 */
@Data
public class PageResult<T> {
    private List<T> items;
    private int total;
    private int pageIndex;
    private int pageSize;

    public PageResult(){
        this.items = new ArrayList<>();
    }


    public PageResult(int pageIndex, int pageSize, int total, List<T> items){
        this.pageIndex = pageIndex;
        this.pageSize = pageSize;
        this.total = total;
        this.items = items;
    }
}

File: server-dotNet/Com.EnuoCms.Core/Common/PageResult.cs
using System;
using System.Collections.Generic;

/**
 * Created by zhanxiaoping on 2017/8/28.
 * zhanxp@me.com
 */
namespace Com.EnuoCms.Core.Common
{
    public class PageResult<T>
    {
        public int pageIndex { get; set; }
        public int pageSize { get; set; }
        public int total { get; set; }
        public List<T> items { get; set; }
      
        public PageResult()
    	{
    		this.items = new List<T>();
    	}

        public PageResult(int pageIndex, int pageSize, int total, List<T> items)
        {
            this.pageIndex = pageIndex;
            this.pageSize = pageSize;
            this.total = total;
            this.items = items;
        }
    }
}
